pkg/ibmvpc/sg_analysis.go

```diff
@@ -70,35 +107,19 @@ func (sga *SGAnalyzer) getRemoteCidr(remote vpc1.SecurityGroupRuleRemoteIntf) (*
 		switch {
 		case remoteObj.CIDRBlock != nil:
 			cidr = *remoteObj.CIDRBlock
-			target, err = ipblock.FromCidr(cidr)
-			if err != nil {
-				return nil, "", err
-			}
-			cidrRes = target.ToCidrList()[0]
 		case remoteObj.Address != nil:
-			address := *remoteObj.Address
-			target, err = ipblock.FromIPAddress(address)
-			if err != nil {
-				return nil, "", err
-			}
-			cidrRes = target.ToCidrList()[0]
+			address = *remoteObj.Address
 		case remoteObj.Name != nil:
-			if remoteSg, ok := sga.sgMap[*remoteObj.Name]; ok {
-				resIPBlock := ipblock.New()
-				for member := range remoteSg.members {
-					memberIPBlock, err := ipblock.FromIPAddress(member)
-					if err != nil {
-						return nil, "", err
-					}
-					resIPBlock = resIPBlock.Union(memberIPBlock)
-				}
-				target = resIPBlock
-				cidrRes = strings.Join(target.ToCidrList(), ",")
-			}
+			name = *remoteObj.Name
 		default:
 			return nil, "", fmt.Errorf("sg error: getRemoteCidr - SecurityGroupRuleRemote is empty: %+v", remoteObj)
 		}
 
+		target, cidrRes, err = sga.getIPBlockResult(cidr, address, name)
```

Comment:
`getIPBlockResult()` can get args of type `*string`, and then no need for the switch above